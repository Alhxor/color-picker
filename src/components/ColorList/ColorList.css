ul {
    list-style-type: none;
    display: inherit;
    align-items: inherit;
    justify-content: inherit;
    width: 80%;
}

li {
    margin: 0.2em 1em;
    display: inherit;
    flex-direction: column;
    align-items: inherit;
    justify-content: inherit;
}
