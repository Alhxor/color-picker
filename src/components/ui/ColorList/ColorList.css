ul {
    list-style-type: none;
    display: flex;
    flex-flow: wrap;
    align-items: center;
    justify-content: center;
    width: 80%;
    margin: 0;
    padding: 0;
}

li {
    margin: 0.2em 1em;
    display: inherit;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}
