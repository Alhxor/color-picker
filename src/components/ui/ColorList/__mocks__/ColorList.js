const ColorListMock = ({ colors, rateColor }) => (
    <ul className="color-list-mock">{
        colors.map(c => (
            <li className="color" key={c.id}>
                <span className="title">{c.title}</span>
                <span className="hexCode">{c.hexCode}</span>
                <button className="rate" onClick={rateColor}>rate</button>
            </li>
        ))
    }</ul>
);

ColorListMock.displayName = 'ColorListMock';

export default ColorListMock;
