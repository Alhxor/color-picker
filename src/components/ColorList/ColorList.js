import React from 'react';
import PropTypes from 'prop-types';
import Color from '../Color/Color';
import StarRating from '../StarRating/StarRating';
import './ColorList.css';

const ColorList = ({ colors, onColorRatingChange = (...x) => x }) => (
    <ul>
        {colors.map((color, i) => (
            <li key={color.id}>
                <Color
                    name={color.name}
                    hexCode={color.hexCode}
                />
                <StarRating
                    totalStars={5}
                    starsSelected={color.rating}
                    onRatingChange={onColorRatingChange(color.id)}
                />
            </li>
        ))}
    </ul>
);

ColorList.propTypes = {
    colors: PropTypes.array.isRequired,
    onColorRatingChange: PropTypes.func
};

export default ColorList;
