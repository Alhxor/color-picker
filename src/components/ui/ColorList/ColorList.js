import React from 'react';
import PropTypes from 'prop-types';
import Color from '../Color/Color';
import StarRating from '../StarRating/StarRating';
import './ColorList.css';

const ColorList = ({
    colors,
    rateColor = (...x) => x,
    removeColor = (...x) => x
}) => (
    <ul>
        {colors.map(color => {
            console.log(color);
            return (
            <li key={color.id}>
                <Color
                    title={color.title}
                    hexCode={color.hexCode}
                    removeColor={removeColor(color.id)}
                />
                <StarRating
                    totalStars={5}
                    starsSelected={color.rating}
                    onRatingChange={rateColor(color.id)}
                />
            </li>
        )})}
    </ul>
);

ColorList.propTypes = {
    colors: PropTypes.array.isRequired,
    rateColor: PropTypes.func,
    removeColor: PropTypes.func
};

export default ColorList;
