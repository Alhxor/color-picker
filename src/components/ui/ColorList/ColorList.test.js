import { shallow, mount } from 'enzyme';
import ColorList from './ColorList';

jest.mock('../StarRating/StarRating');

describe("<ColorList /> ui component", () => {
    let wrapper;
    const _rate = jest.fn()
    beforeAll(() => {
        wrapper = mount(<ColorList colors={_testColors} rateColor={_rate} />);
    });

    it("renders without crashing", () =>
        expect(wrapper.exists()).toBeTruthy());

    it("renders three colors", () =>
        expect(wrapper
            .find('ul')
            .children()
        ).toHaveLength(3));

    it("calls rateColor on button click", () => {
        wrapper
            .find('button.rate')
            .at(0)
            .simulate('click')
        expect(_rate).toHaveBeenCalled();
    });

    it("matches snapshot", () =>
        expect(wrapper).toMatchSnapshot());
});
