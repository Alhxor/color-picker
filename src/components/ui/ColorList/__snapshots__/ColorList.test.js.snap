// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorList /> ui component matches snapshot 1`] = `
<ColorList
  colors={
    Array [
      Object {
        "hexCode": "#aa0000",
        "id": "testred",
        "rating": 1,
        "title": "red",
      },
      Object {
        "hexCode": "#00aa00",
        "id": "testgreen",
        "rating": 4,
        "title": "green",
      },
      Object {
        "hexCode": "#0000aa",
        "id": "testblue",
        "rating": 3,
        "title": "blue",
      },
    ]
  }
  rateColor={
    [MockFunction] {
      "calls": Array [
        Array [
          "testred",
        ],
        Array [
          "testgreen",
        ],
        Array [
          "testblue",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <ul>
    <li
      key="testred"
    >
      <Color
        hexCode="#aa0000"
        removeColor={
          Array [
            "testred",
          ]
        }
        title="red"
      >
        <div
          className="show-color"
        >
          <span
            className="color-title"
          >
            red
          </span>
          <span
            className="remove-color"
            onClick={
              Array [
                "testred",
              ]
            }
          >
            <FaTrash>
              <IconBase
                attr={
                  Object {
                    "viewBox": "0 0 448 512",
                  }
                }
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                    key="0"
                  />
                </svg>
              </IconBase>
            </FaTrash>
          </span>
          <div
            className="color-sample"
            style={
              Object {
                "backgroundColor": "#aa0000",
              }
            }
          />
        </div>
      </Color>
      <StarRatingMock
        starsSelected={1}
        totalStars={5}
      >
        <div
          className="star-rating-mock"
        >
          <span
            className="stars"
          />
          <span
            className="selected"
          />
          <button
            className="rate"
          >
            rate
          </button>
        </div>
      </StarRatingMock>
    </li>
    <li
      key="testgreen"
    >
      <Color
        hexCode="#00aa00"
        removeColor={
          Array [
            "testgreen",
          ]
        }
        title="green"
      >
        <div
          className="show-color"
        >
          <span
            className="color-title"
          >
            green
          </span>
          <span
            className="remove-color"
            onClick={
              Array [
                "testgreen",
              ]
            }
          >
            <FaTrash>
              <IconBase
                attr={
                  Object {
                    "viewBox": "0 0 448 512",
                  }
                }
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                    key="0"
                  />
                </svg>
              </IconBase>
            </FaTrash>
          </span>
          <div
            className="color-sample"
            style={
              Object {
                "backgroundColor": "#00aa00",
              }
            }
          />
        </div>
      </Color>
      <StarRatingMock
        starsSelected={4}
        totalStars={5}
      >
        <div
          className="star-rating-mock"
        >
          <span
            className="stars"
          />
          <span
            className="selected"
          />
          <button
            className="rate"
          >
            rate
          </button>
        </div>
      </StarRatingMock>
    </li>
    <li
      key="testblue"
    >
      <Color
        hexCode="#0000aa"
        removeColor={
          Array [
            "testblue",
          ]
        }
        title="blue"
      >
        <div
          className="show-color"
        >
          <span
            className="color-title"
          >
            blue
          </span>
          <span
            className="remove-color"
            onClick={
              Array [
                "testblue",
              ]
            }
          >
            <FaTrash>
              <IconBase
                attr={
                  Object {
                    "viewBox": "0 0 448 512",
                  }
                }
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                    key="0"
                  />
                </svg>
              </IconBase>
            </FaTrash>
          </span>
          <div
            className="color-sample"
            style={
              Object {
                "backgroundColor": "#0000aa",
              }
            }
          />
        </div>
      </Color>
      <StarRatingMock
        starsSelected={3}
        totalStars={5}
      >
        <div
          className="star-rating-mock"
        >
          <span
            className="stars"
          />
          <span
            className="selected"
          />
          <button
            className="rate"
          >
            rate
          </button>
        </div>
      </StarRatingMock>
    </li>
  </ul>
</ColorList>
`;
